### ISSUE TITLE
Bug: schemas of some rules allow extra properties

### ISSUE THREAD

**Issue by andreww2012** [typescript-eslint/typescript-eslint#11396]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

-

### Repro Code

```TypeScript
-
```

### ESLint Config

```javascript
-
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Some rules, for example [prefer-destructuring](https://github.com/typescript-eslint/typescript-eslint/blob/16c344ec7d274ea542157e0f19682dd1930ab838/packages/eslint-plugin/src/rules/prefer-destructuring.ts#L53) allow extra properties to be passed in options object, which should not be allowed.

While not strictly a bug, this makes it easier for typos in rule options to go unnoticed.

### Actual Result

Extra properties are allowed.

**Comment by cseas** [comment]
Looks like this is fixed already in #11397
